frontend-dev/src/components/Utilities/UserRadioButton.jsx

```diff
@@ -0,0 +1,103 @@
+import React, { useState, useEffect, useRef } from 'react';
+import bitsFetch from '../../Utils/bitsFetch';
+
+const UserRadioButton = ({ authData, setAuthData, selectedUser, setSelectedUser, handleAuthUser, isInfo, setIsLoading }) => {
+  const [showConfirm, setShowConfirm] = useState(null);
+  const popoverRef = useRef(null); // Ref to the popover container
+
+  const handleDelete = (id) => {
+    const index = authData.findIndex(item => item.id === id.toString());
+    const newAuthData = authData.filter(item => item.id !== id);
+    setIsLoading(true)
+    bitsFetch(id, 'auth/account/delete').then((res) => {
+      if (res.success) {
+        setAuthData(newAuthData);
```

Comment:
- pass the index in the `handleDelete` params
- use the functional update pattern to filter newAuthData directly in the setAuthData function